pkg/redisx/pool.go

```diff
@@ -43,3 +51,107 @@ func newInstancePool(config etc.RedisPool) *redis.Pool {
 		Wait:        true,
 	}
 }
+
+// redis+sentinel://user:password@sentinel_host1:port1,sentinel_host2:port2/monitor-name/db-number
+func newSentinelPool(configURL *url.URL, config etc.RedisPool) (pool *redis.Pool, err error) {
+	log.Trace("Constructing connection pool for Redis Sentinel")
+	sentinelURL, err := ParseSentinelURL(configURL)
+	if err != nil {
+		return
+	}
+
+	var commonOpts []redis.DialOption
+	if config.ConnectionTimeout > 0 {
+		commonOpts = append(commonOpts, redis.DialConnectTimeout(config.ConnectionTimeout))
+	}
+	if config.ReadTimeout > 0 {
+		commonOpts = append(commonOpts, redis.DialReadTimeout(config.ReadTimeout))
+	}
+	if config.WriteTimeout > 0 {
+		commonOpts = append(commonOpts, redis.DialWriteTimeout(config.WriteTimeout))
+	}
+
+	sentinelOpts := commonOpts
+
+	sntnl := &sentinel.Sentinel{
+		Addrs:      sentinelURL.Addrs,
+		MasterName: sentinelURL.MonitorName,
+		Dial: func(addr string) (conn redis.Conn, err error) {
+			log.WithField("addr", addr).Trace("Connecting to Redis sentinel")
+			conn, err = redis.Dial("tcp", addr, sentinelOpts...)
+			if err != nil {
+				return
+			}
+			return
+		},
+	}
+
+	redisOpts := commonOpts
+
+	if db := sentinelURL.Database; db != 0 {
+		redisOpts = append(redisOpts, redis.DialDatabase(db))
+	}
```

Comment:
nit: IIRC, the default database is 0. Do we need this if condition?

```suggestion
	redisOpts = append(redisOpts, redis.DialDatabase(db))
```

Reply:
You're right! We can simplify that. Thank you!